src/lets_be_rational/rational_cubic.rs

```diff
@@ -0,0 +1,183 @@
+use num_traits::Zero;
+
+const MINIMUM_RATIONAL_CUBIC_CONTROL_PARAMETER_VALUE: f64 = -(1.0 - 1.4901161193847656e-8); // -(1.0 - f64::EPSILON.sqrt());
+const MAXIMUM_RATIONAL_CUBIC_CONTROL_PARAMETER_VALUE: f64 = 2.0 / (f64::EPSILON * f64::EPSILON);
+
+pub fn rational_cubic_interpolation(
+    x: f64,
+    x_l: f64,
+    x_r: f64,
+    y_l: f64,
+    y_r: f64,
+    d_l: f64,
+    d_r: f64,
+    r: f64,
+) -> f64 {
+    let h = x_r - x_l;
+    if h.abs() <= 0.0 {
+        return 0.5 * (y_l + y_r);
+    }
+    // r should be greater than -1.
+    // We do not use assert(r > -1) here in order to allow values such as NaN to be propagated as they should.
```

Comment:
in my opinion these implementation of cubic is terrible...

Reply:
I think we should merge as is and look to replace this later if improvements can be made